### PR TITLE
[PLATFORM-430] Right-click context menu for ports

### PR DESCRIPTION
**Pull request by tumppi**
Adds right click context menu for ports. You can disconnect all connections from the port or toggle the export on/off.

### CODE REVIEW

**Comment by tumppi** [comment]
Thanks for the comments. I'll check what we can do to improve event handling 👍

**Comment by tumppi** [comment]
Improved event handling based on Tim's comments and also added storybook story 👍

File: app/src/shared/components/ContextMenu/index.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    static Item = ContextMenuItem
+
+    render = () => {
+        const { target, placement, isOpen, children } = this.props
+        return (
+            <Popover
+                isOpen={isOpen}
+                placement={placement}
+                target={target}
+                className={styles.menu}
```
Comment (timoxley) [inline app/src/shared/components/ContextMenu/index.jsx]: Always good to allow components to be passed custom `className` prop, i.e. `className={cx(props.className, styles.menu)}`

File: app/src/editor/canvas/components/Ports.jsx [hunk 2]
```diff
@@ -57,10 +57,75 @@ const getPortDisplayValueSize = ({ value, defaultValue, possibleValues }) => {
...
+        this.setState({
+            isMenuOpen: true,
+        })
+    }
+
+    closeContextMenu = () => {
+        // Hide with a delay so that ContextMenuItem has time to
+        // react to click event before element unmounts.
+        setTimeout(() => {
+            if (this.unmouted) { return }
```
Comment (timoxley) [inline app/src/editor/canvas/components/Ports.jsx]: `unmouted`

File: app/src/shared/components/ContextMenu/ContextMenuItem/index.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        }
+    }
+
+    render = () => {
+        const { text } = this.props
+        return (
+            // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+            <div
+                role="button"
+                className={styles.item}
```
Comment (timoxley) [inline app/src/shared/components/ContextMenu/ContextMenuItem/index.jsx]: Permit className prop

**Comment by timoxley** [comment]
@tumppi can you rebase this PR on `development` please? This branch seems to be missing the docker env travis scripts and it's likely that CI is failing due to that.

**Comment by tumppi** [comment]
That did the trick @timoxley 👍

**Pull request merged**